### ISSUE TITLE
Pass xRoute Spec object instead of entire xRoute object using watchable 

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#1370]
*Description*:
The xRoutes are passed from the `provider` to the `gateway-api` layer using the watchable messaging system.
https://github.com/envoyproxy/gateway/blob/a5d3f36227b4c48f9dd70a60b213bd5278991e1f/internal/message/types.go#L23
Instead of passing the [entire resources](https://github.com/envoyproxy/gateway/blob/a5d3f36227b4c48f9dd70a60b213bd5278991e1f/internal/gatewayapi/resource.go#L25) we can only pass the [Spec](https://github.com/kubernetes-sigs/gateway-api/blob/16c54cdd2546bce917c1d045a5dc6a49e51625ae/apis/v1beta1/httproute_types.go#L56) which is the required for translation.

This should reduce memory consumption as well as improve caching hits, since any field related to `time` that is part of the meta resource or status will not affect the object comparison and will reduce object churn in the system

Relates to https://github.com/envoyproxy/gateway/issues/1365

**Comment by haq204** [comment]
We  use `metadata.name` on some resources for xDS names

**Comment by arkodg** [comment]
> We use `metadata.name` on some resources for xDS names https://github.com/envoyproxy/gateway/blob/main/internal/xds/translator/route.go#L23

good point, we'll then need to find to only share the Namespace, Name and Spec
